staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader.go

```diff
@@ -184,6 +174,23 @@ func (a *requestHeaderAuthRequestHandler) AuthenticateRequest(req *http.Request)
 	}, true, nil
 }
 
+func ClearAuthenticationHeaders(h http.Header, nameHeaders, groupHeaders, extraHeaderPrefixes StringSliceProvider) {
+	for _, headerName := range nameHeaders.Value() {
+		h.Del(headerName)
+	}
+	for _, headerName := range groupHeaders.Value() {
+		h.Del(headerName)
+	}
+	for _, prefix := range extraHeaderPrefixes.Value() {
+		prefix := strings.ToLower(prefix)
```

Comment:
ToLower allocates, I'd suggest a hasPrefixIgnoreCase helper that checks lengths, then calls strings.EqualFold on substring matching the length of the prefix check

Reply:
Done.